src/ripple/rpc/impl/RPCHelpers.cpp

```diff
@@ -117,7 +117,18 @@ getAccountObjects(ReadView const& ledger, AccountID const& account,
         for (; iter != entries.end (); ++iter)
         {
             auto const sleNode = ledger.read(keylet::child(*iter));
-            if (type == ltINVALID || sleNode->getType () == type)
+
+            auto typeMatchesFilter = [] (
+                std::vector<LedgerEntryType> const& typeFilter,
+                LedgerEntryType ledgerType)
+            {
+                auto it = std::find(typeFilter.begin(), typeFilter.end(),
+                    ledgerType);
+                return it != typeFilter.end();
+            };
+
+            if (typeFilter.empty() ||
```

Comment:
I'd expect an empty vector to return no values, not all the values. Since this is part of the interface to this function, I think we should clarify the interface. The two ways I can think of are:

1) Make `typeFilter` an `optional<vector>`, and an empty optional means "no filter".
2) Make `typeFilter` a bitset that can be indexed by an ascii character. The "no filter" case would be handled by setting all the bits in this bitset.